### PR TITLE
Add DotNet 8.0 Support

### PR DESCRIPTION
**Pull request by BroMarduk**
- Added DotNet 8.0 Libraries
- Rewrote some functions to support previous versions of frameworks for Utilites (support back to NetStandard2.0)

### CODE REVIEW

File: OpenAI.UtilitiesPlayground/TestHelpers/FunctionCallingTestHelpers.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ public static async Task ExerciseFunctionCalling(IOpenAIService openAIService)
         {
             //Functions = FunctionCallingHelper.GetFunctionDefinitions(calculator),
             //Functions = FunctionCallingHelper.GetFunctionDefinitions(typeof(Calculator)),
-            Functions = FunctionCallingHelper.GetFunctionDefinitions<Calculator>(),
+            //Functions = FunctionCallingHelper.GetFunctionDefinitions<Calculator>(),
```
Comment (BroMarduk) [inline OpenAI.UtilitiesPlayground/TestHelpers/FunctionCallingTestHelpers.cs:18]: Functions no longer seems to exist...

File: OpenAI.Utilities/Embedding/EmbeddingTools.cs [hunk 2]
```diff
@@ -200,7 +202,7 @@ public DataFrame LoadEmbeddedDataFromCsv(string path)
     /// <returns></returns>
     public IEnumerable<TextEmbeddingData> LoadFilesFromDirectory(string pathToDirectory)
     {
-        return !Path.Exists(pathToDirectory)
+        return !File.Exists(pathToDirectory)
```
Comment (BroMarduk) [inline OpenAI.Utilities/Embedding/EmbeddingTools.cs:205]: Path.Exists not supported in < .NET 7.0

File: OpenAI.Utilities/Embedding/EmbeddingTools.cs [hunk 3]
```diff
@@ -1,4 +1,6 @@
-﻿using System.Collections;
+﻿#if NET6_0_OR_GREATER
```
Comment (BroMarduk) [inline OpenAI.Utilities/Embedding/EmbeddingTools.cs:1]: This utility will not run with .NetStandard 2.0 without more fixes so excluding it from utilities for now.

File: OpenAI.Utilities/FunctionCalling/FunctionCallingHelper.cs [hunk 4]
```diff
@@ -153,7 +153,7 @@ public static List<FunctionDefinition> GetFunctionDefinitions(Type type)
             throw new InvalidFunctionCallException($"Method '{functionCall.Name}' on type '{obj.GetType()}' not found");
         }
 
-        if (!methodInfo.ReturnType.IsAssignableTo(typeof(T)))
+        if (!typeof(T).IsAssignableFrom(methodInfo.ReturnType))
```
Comment (BroMarduk) [inline OpenAI.Utilities/FunctionCalling/FunctionCallingHelper.cs:156]: IsAssignableTo not supported in .NetStandard 2.0 so reversed it to IsAssignableFrom for support.

**Comment by kayhantolga** [comment]
I have reverted the changes in the utilization library. The utilization library does not support, and will not support, legacy versions, including .NET 6 and .NET 7.

**Pull request merged**